File: types/product.ts
export type Product = {
  id: number;
  title: string;
  price: number;
  description: string;
  category: string;
  image: string;
  rating:{
    rate: number;
    count: number;
  }
}

export type ProductListProps ={
    data: Product[];
    page: number;
    limit: number;
    total: number
}

File: backend/src/domain/product.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Product {
    pub id: u32,
    pub title: String,
    pub price: f64,
    pub description: String,
    pub category: String,
    pub image: String,
    pub rating: Rating,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Rating {
    pub rate: f64,
    pub count: u32,
}
